core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -558,4 +558,32 @@ default ClientRequestContext unwrap() {
     default ClientRequestContext unwrapAll() {
         return (ClientRequestContext) RequestContext.super.unwrapAll();
     }
+
+    /**
+     * Returns the authority which will eventually be sent when a {@link Client} sends an {@link HttpRequest}.
+     * This method checks the following locations and returns the first non-null value.
+     * <ol>
+     *     <li>Either the {@link HttpHeaderNames#HOST} or {@link HttpHeaderNames#AUTHORITY} value from
+     *         {@link ClientRequestContext#additionalRequestHeaders()}.</li>
+     *     <li>The {@link HttpRequest#authority()} from {@link ClientRequestContext#request()}.</li>
+     *     <li>{@link ClientRequestContext#defaultRequestHeaders()}.</li>
+     *     <li>{@link Endpoint#authority()}.</li>
+     * </ol>
+     */
+    @Nullable
+    String authority();
+
+    /**
+     * Returns the uri constructed based on {@link ClientRequestContext#sessionProtocol()},
+     * {@link ClientRequestContext#authority()}, {@link ClientRequestContext#path()} and
+     * {@link ClientRequestContext#query()}.
```

Comment:
Should we also add `#fragment`? If we don't, we probably should make it clear.

Reply:
We should 😅    I must have missed this

Reply:
Just by receiving a `HttpRequest` I think it's difficult to determine whether 1) the path actually contains a `#` character 2) or the uri contains a fragment.

Rather than guessing, I think `fragment` follows the lifecycle of `ClientRequestContext` rather than `HttpRequest`.
Even now, there is no way to explicitly set a `fragment` to `HttpRequest`, and `THttpClient` sets the fragment to `DefaultTHttpClient` instead.

https://github.com/line/armeria/blob/97a17cb1e6671a8b73a61b45b3e460784adcf214/thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/client/thrift/DefaultTHttpClient.java#L77-L78

I think it's reasonable behavior to assume that `fragment` won't be updated when `ClientRequestContext#updateRequest` is invoked. Let me update the javadocs to make this clear. Thanks for pointing this out 🙇